asserts/snapasserts/validation_sets.go

```diff
@@ -449,3 +449,68 @@ func (v *ValidationSets) CheckInstalledSnaps(snaps []*InstalledSnap) error {
 	}
 	return nil
 }
+
+func (v *ValidationSets) constraintsForSnap(snapRef naming.SnapRef) *snapContraints {
+	if snapRef.ID() != "" {
+		return v.snaps[snapRef.ID()]
+	}
+	// snapID not available, find by snap name
+	for _, cstrs := range v.snaps {
+		if cstrs.name == snapRef.SnapName() {
+			return cstrs
+		}
+	}
+	return nil
+}
+
+// CheckPresenceRequired returns the list of all validation sets that declare
+// presence of the given snap as required and the required revision (or
+// snap.R(0) if no specific revision is required).
+// The method assumes that validation sets are not in conflict.
+func (v *ValidationSets) CheckPresenceRequired(snapRef naming.SnapRef) ([]string, snap.Revision) {
```

Comment:
I think we might want to return an error from this and have an error if presence is Invalid

Reply:
Done.